tools/autocorres/L1Defs.thy

```diff
@@ -334,7 +334,7 @@ lemma no_throw_L1_init [simp]: "no_throw P (L1_init f)"
   apply (rule no_throw_bindE [where B=\<top>])
     apply simp
    apply simp
-  apply wp
+  apply wpsimp
```

Comment:
any thoughts on why the switch to wpsimp was required?

Reply:
I'd have to go in to check this lemma specifically, but generally that switch was needed when the proof used to work with some rule that could directly solve the hoare triple, instead of the more general `*_wp`  which leaves behind a final goal. I guess an alternative would be to explicitly add that other rule again here so that it is used first, but that feels more clunky.

Reply:
Got it, thanks for the explanation. No need to dive back in.